### ISSUE TITLE
Cleanup usage of kubernetes-release-pull in kubernetes presubmits

### ISSUE THREAD

**Issue by amwat** [kubernetes/test-infra#18789]
**What should be cleaned up or changed**:
We stage builds to `gs://kubernetes-release-pull` in almost every presubmit job.
But from what I can tell nothing is actually consuming those builds since the jobs also use `extract=local` .
It's a non-trivial overhead to upload the release tars in every presubmit and we should remove all the non-required usages.

**Provide any links for context**:
https://cs.k8s.io/?q=kubernetes-release-pull&i=nope&files=&repos=
https://github.com/kubernetes/test-infra/blob/c4628a3a1c2e4c149f669348f351fe78ebd1f258/kubetest/extract_k8s.go#L449-L466
Random GCE provider job: https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/directory/pull-kubernetes-e2e-gce/1293275406807339008#1:build-log.txt%3A903

/cc @spiffxp @BenTheElder @MushuEE 

---
EDIT(@spiffxp): I made a list of the offending jobs going off the criteria `--extract=local` and `--stage=gs://kubernetes-release-pull/*`
- if the job triggers for a single branch it's labeled as `job@branch`
- if the job triggers for all branches it's labeled as `job`
- there are no presubmits that trigger for N branches (where all > N > 1)
- there are no periodics or postsubmits that touch gs://kubernetes-release-pull
- this picks up some `--provider=aws` jobs (kops), it remains to be seen whether they need `--stage` or not

EDIT(@BenTheElder): I removed the outdated checklist and instead i'm going to provide a search: https://github.com/search?q=repo%3Akubernetes%2Ftest-infra+%22--stage%3Dgs%3A%2F%2Fkubernetes-release-pull%22&type=code

**Comment by BenTheElder** [comment]
we should test this in a canary just because this stuff is old and brittle and I can't remember why we were doing this anymore 🙃

**Comment by amwat** [comment]
update: 
https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/92316/pull-kubernetes-e2e-gce-no-stage/1294345203125063682/#1:build-log.txt%3A355

the local path
https://github.com/kubernetes/test-infra/blob/c4628a3a1c2e4c149f669348f351fe78ebd1f258/kubetest/extract_k8s.go#L450

comes from 
https://github.com/kubernetes/release/blob/8d6bd15010efeec44018e4847860d464d2682d97/lib/releaselib.sh#L1245-L1247

which shouldn't be needed since we already have them under 
`bazel-bin/build/release-tars` (but without the hashes)
https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/92316/pull-kubernetes-e2e-gce-no-stage/1294345203125063682/#1:build-log.txt%3A338

**Comment by BenTheElder** [comment]
still worth doing?

**Comment by spiffxp** [comment]
/remove-lifecycle rotten
I think so.  The other option is to continue as-is, meaning jobs that use this bucket need to switch to use k8s-release-pull as they migrate to k8s-infra.

**Comment by amwat** [comment]
sadly looks like those gcs links have been gced.
seems like one of the steps involved as part of `--stage` is copying the artifacts from the bazel output path to the make output path `_output/gcs-stage` and then uploading them to gcs.

and our presubmit jobs are configured to `--extract=local` instead of `--extract=bazel` while using `--build=bazel`
so they were relying on them being in the make output path.
https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes/sig-cloud-provider/gcp/gcp-gce.yaml#L48

testing out in the canary job: https://github.com/kubernetes/test-infra/pull/20427

**Comment by amwat** [comment]
We have a succesful run at https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/directory/pull-kubernetes-e2e-gce-no-stage/1352340847076577280

not sure why the total test duration is higher as compared to 
https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/directory/pull-kubernetes-e2e-gce/1351850610516824064

but we atleast saved 154 seconds of staging time and 1.1 (which should be the only delta here)

https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/92316/pull-kubernetes-e2e-gce-no-stage/1352340847076577280/artifacts/junit_runner.xml 

as compared to 

https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/97894/pull-kubernetes-e2e-gce/1351850610516824064/artifacts/junit_runner.xml

and 1.84 GiB of unnecessary GCS uploads 

```
gsutil du -sh gs://kubernetes-release-pull/ci/pull-kubernetes-e2e-gce/v1.18.16-rc.0.3+9f5c61d324a62b
```

**Comment by spiffxp** [comment]
/assign @amwat @spiffxp 
Assigning to us for now.  If we think this is eligible for /help or don't have time to do it ourselves we can writeup how to proceed

**Comment by amwat** [comment]
(either way) **Next steps:**
All jobs in question:
https://cs.k8s.io/?q=kubernetes-release-pull&i=nope&files=&repos= i.e. those with `--stage=gs://kubernetes-release-pull.*`

affected jobs: those with `--provider=gce` `--extract=local` `--build=bazel` 
should be fixed to have `--extract=bazel` and no `--stage`

short of having a canary job for every affected job
a reasonable, while having less disruption, compromise would be to start with those that are non-blocking
e.g. https://testgrid.k8s.io/presubmits-kubernetes-nonblocking#pull-kubernetes-e2e-gce-alpha-features
and once they are monitored to be healthy, fix the blocking jobs.

**Comment by spiffxp** [comment]
Updated the description with a list of jobs to update as generated by the test I'm adding in https://github.com/kubernetes/test-infra/pull/22890

```shell
go test -v -count=1 \
  ./config/tests/jobs/ \
  -run TestKubernetesPresubmitsShouldNotUseKubernetesReleasePullBucket \
| grep "should not" \
| cut -d: -f4 \
| sort | uniq \
| sed -e 's/^/- [ ]/'
```

**Comment by spiffxp** [comment]
Opened https://github.com/kubernetes/test-infra/pull/22892 to go after that jobs that will be most obvious if this somehow ends up having an unforeseen negative impact

**Comment by spiffxp** [comment]
https://github.com/kubernetes/test-infra/pull/22892 broke things, ref: https://github.com/kubernetes/test-infra/pull/22892#issuecomment-880245096

>The analysis in #18789 (comment) depended on bazel doing the local staging, but now that there's no bazel we seem to be still relying on just the local staging part of the --stage for --extract=local.

Reverted in https://github.com/kubernetes/test-infra/pull/22894

**Comment by spiffxp** [comment]
/milestone v1.23
Let's just wait until v1.22 goes out the door before we bother with this again

**Comment by BenTheElder** [comment]
/lifecycle frozen
this is a requirement for smooth migration
/help

**Comment by jbpratt** [comment]
Based on https://github.com/kubernetes/test-infra/pull/22892#issuecomment-880245096 (and the changes being reverted), how should we proceed with this? I started working through this and realized I was re-doing @spiffxp's changes :smile:

**Comment by SD-13** [comment]
Hi, I am interested to work on this issue but I have some questions or queries.
1. Seems like the list of jobs to be fixed is outdated
2. Please help me to understand the fix we need to follow here
I don't think the fix @amwat mentioned [here](https://github.com/kubernetes/test-infra/issues/18789#issuecomment-768004759) here don't apply because we are not using `bazel` to build.
So as discussed here https://github.com/kubernetes/test-infra/pull/24238#issuecomment-961504149, can we now remove extract and stage?
Please feel free to correct me if I am wrong

**Comment by SD-13** [comment]
Hi, I am interested to work on this issue but I have some questions or queries.
1. Seems like the list of jobs to be fixed is outdated
2. Please help me to understand the fix we need to follow here
I don't think the fix @amwat mentioned [here](https://github.com/kubernetes/test-infra/issues/18789#issuecomment-768004759) here don't apply because we are not using `bazel` to build.
So as discussed here https://github.com/kubernetes/test-infra/pull/24238#issuecomment-961504149, can we now remove extract and stage?
Please feel free to correct me if I am wrong

cc @spiffxp @BenTheElder @ameukam

**Comment by BenTheElder** [comment]
Sorry, a couple of the people you pinged don't work on tthis anymore and I'm kinda buried.

I've lost context on this one.

**Comment by BenTheElder** [comment]
I'm not sure we ever got no-stage working? It's hard to follow at this point.

**Comment by BenTheElder** [comment]
https://github.com/kubernetes/test-infra/pull/28176 renamed the test job, testing in https://github.com/kubernetes/kubernetes/pull/126563

**Comment by BenTheElder** [comment]
It does, it will stage to a generated bucket under the rented boskos project (which the boskos janitors should clean up if they don't already), so we can carefully start dropping these I think ... very belatedly.

**Comment by BenTheElder** [comment]
beginning bulk migration in https://github.com/kubernetes/test-infra/pull/33259, starting with a subset of optional, non-blocking, not always_run jobs

We have to drop both `--extract=local` and `--stage` at the same time. We don't need to locally extract what we just built, it's running fine and uploading to a bucket under the boskos project.

You can see sample runs in https://github.com/kubernetes/kubernetes/pull/126563

Inspect these logs:
https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/126563/pull-kubernetes-e2e-gce-cos-no-stage/1820909457454927872
https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/126563/pull-kubernetes-e2e-gce-pull-through-cache/1821254221408768000

**Comment by BenTheElder** [comment]
https://github.com/kubernetes/test-infra/pull/33278 does everything but the one remaining PR blocking job, for which we'll wait a bit and check some more things

**Comment by BenTheElder** [comment]
Once we have test results we can do https://github.com/kubernetes/test-infra/pull/33280, and then I'll delete the bucket

**Comment by BenTheElder** [comment]
This is done, I just need to follow-up with eliminating that bucket.

**Comment by BenTheElder** [comment]
Done!
